src/runtime/server/middleware/session/index.ts

```diff
@@ -54,15 +108,19 @@ export const deleteSession = async (event: H3Event) => {
 }
 
 const newSession = async (event: H3Event) => {
-  // Cleanup old session data to avoid leaks
-  await deleteSession(event)
+  const runtimeConfig = useRuntimeConfig()
+  const sessionOptions = runtimeConfig.session.session
 
   // (Re-)Set cookie
-  const sessionId = nanoid(useRuntimeConfig().session.session.idLength)
+  const sessionId = nanoid(sessionOptions.idLength)
   safeSetCookie(event, SESSION_COOKIE_NAME, sessionId)
 
   // Store session data in storage
-  const session: Session = { id: sessionId, createdAt: new Date() }
+  const session: Session = {
+    id: sessionId,
+    createdAt: new Date(),
+    ip: sessionOptions.ipPinning ? await hashIpAddress(getRequestIpAddress(event)) : undefined
```

Comment:
Nit: I would move this logic into a function that lives inside `ip-pinning.ts` as well, eg:
```suggestion
    ip: sessionOptions.ipPinning ? await getHashedIpAddress(event) : undefined,
```